transaction/_manager.py

```diff
@@ -190,29 +191,30 @@ def run(self, func=None, tries=3):
         name = text_(name) if name else u''
         doc = text_(doc) if doc else u''
 
-        if name != u'_':
+        if name and name != u'_':
             if doc:
                 doc = name + u'\n\n' + doc
             else:
                 doc = name
-
-        for i in range(1, tries + 1):  # pragma: no branch
+        
+        for try_no in itertools.count(1):
             txn = self.begin()
             if doc:
                 txn.note(doc)
-
             try:
                 result = func()
-                txn.commit()
-            except Exception as v:
-                if i == tries:
-                    raise # that was our last chance
-                retry = self._retryable(v.__class__, v)
-                txn.abort()
-                if not retry:
-                    raise
-            else:
+                self.commit()
```

Comment:
There's a semantic difference here, right? Previously, if the `commit` action failed, that failure would be raised to the caller. Now if the `commit` actions fails, we attempt to abort (an already broken transaction). That seems right to me, and I assume that was intended. I think, though, that it deserves a specific mention in the change note.